app/src/main/java/de/westnordost/streetcomplete/quests/bike_rental_capacity/AddBikeRentalCapacity.kt

```diff
@@ -0,0 +1,38 @@
+package de.westnordost.streetcomplete.quests.bike_rental_capacity
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.BICYCLIST
+import de.westnordost.streetcomplete.osm.updateWithCheckDate
+
+class AddBikeRentalCapacity : OsmFilterQuestType<Int>() {
+
+    override val elementFilter = """
+        nodes, ways with
+         amenity = bicycle_rental
+         and access !~ private|no
+         and bicycle_rental = docking_station
+         and (!capacity or capacity older today -6 years)
+    """
+
+    override val changesetComment = "Add bicycle rental capacities"
+    override val wikiLink = "Tag:amenity=bicycle_rental"
+    override val icon = R.drawable.ic_quest_bicycle_rental
+    override val isDeleteElementEnabled = true
+    override val questTypeAchievements = listOf(BICYCLIST)
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_bicycle_rental_capacity_title
+
+    override fun getHighlightedElements(element: Element, getMapData: () -> MapDataWithGeometry) =
+        getMapData().filter("nodes, ways with amenity ~ bicycle_parking|bicycle_rental")
```

Comment:
What is the rationale for also showing bicycle parkings here? (In any case, the rationale should be documented as a comment)

Reply:
I think it seems sensible, a drop-off point could look very similar, or even be identical to normal bike parking:
https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbicycle_rental#Types_of_bicycle_rental

Would it make sense to do the reverse in the normal bike quests too?

Reply:
The quest is not asked for dropoff points

Reply:
Ah yes, very true.

I think it's helpful in both directions though to see the other mapped bike parking, to help establish if something has been mis-mapped or removed if you're having to leave a note.

Reply:
The reason why I have been bringing this up is because I think what is shown when highlighting elements must be consistent, i.e. according to what the user would expect. That necessitates that there is a clear guideline what is shown here and what not.
This guideline has not been written down anywhere (yet), but I **think** all quests so far only show same elements that are nearby. E.g. the bench backrest quest just shows nearby benches, but not nearby picnic tables, bleachers or loungers.

Reply:
You're right, with a few (very) minor exceptions:

* `AddBicyclePump` also highlights `compressed_air`, despite even excluding it in the quest
* `AddDrinkingWater` also highlights `amenity=drinking_water`, despite the quest not being asked for those
* `AddPowerPolesMaterial` also highlights power lines

But I guess you're right, let's keep it consistent then.

Reply:
Thank you for checking. For power poles it definitiely makes sense. The question for drinking water is ambiguous enough "water potable here?" that a user may confuse that the question is intended to be asked for a nearby dedicated drinking water fountain, so I think this is fine. For the bicycle pump, I wouldn't mind if this is removed as I guess there is probably no risk for confusion.

Reply:
> The reason why I have been bringing this up is because I think what is shown when highlighting elements must be consistent, i.e. according to what the user would expect. That necessitates that there is a clear guideline what is shown here and what not. This guideline has not been written down anywhere (yet), but I **think** all quests so far only show same elements that are nearby. E.g. the bench backrest quest just shows nearby benches, but not nearby picnic tables, bleachers or loungers.
> 
> Changing this guideline to "show same and elements that are somehow similar and related" I think would not be a good idea because there is no clear line to set what is still similar and what not, so it becomes inconsistent.

I'd argue the reverse, because although it might not be consistent, it should hopefully result in a better quality of data overall.

E.g. the bench quest lets you convert a bench to a picnic table:
https://github.com/streetcomplete/StreetComplete/blob/b0c5853bd0a9df24264b3e7fb2761ea02bbb1739/app/src/main/java/de/westnordost/streetcomplete/quests/bench_backrest/AddBenchBackrest.kt#L40-L43

If there are bench and picnic table nodes nearby on the map, but only a picnic table remaining in real life, without seeing the other related nodes it would be quite easy to end up with two picnic tables.

Likewise OSM having a bike rental and a parking, but the parking having gone in reality, I might leave a confusing note saying this is actually a rental for a quest on the parking node.

Possibly a bit more controversial, bins and benches often go side by side. I've often found and left notes for the micro-mapping of them not being right (e.g. benches on the wrong side of a path, or bench and bin the wrong way round). Apart from the existence quest and a bit of Kim's Game memory, this is hard to do, but would be far simpler if they were shown on the alternate quest.

Essentially because the base SC map is so bare, we potentially miss other opportunities for the boots on the ground to improve the situation further.

Reply:
I agree with @peternewman here - better context leading to better data quality _should trump_ consistency. 

Which is actually why `AddBicyclePump` shows `compressed_air`, even if quest itself fills only `service:bicycle:pump=*` - bicycle shops and repair stations might have manual (i.e. `service:bicycle:pump`) or machine (i.e. `compressed_air`) air sources, and it might be tagged as either, according to individual mapper. Hiding that context from SC user would IMHO lead to lower data quality.

Afterall, we already disregard that consistency in other cases, like in the name of the ease of programming (e.g. `IS_SHOP_OR_DISUSED_SHOP_EXPRESSION` which contains things like `post_office` or `tanning_salon` is shown in quests like `AddHalal` (although `tanning_salon` is not likely to sell any food - Halal or not), instead of showing _only_ restaurants and other shops used in `AddHalal` quest element selection.

Reply:
> AddHalal

Because shop-type elements are shown when selecting a quest about a shop. This behavior is consistent. The highlighted elements are shown to help the user locate the object in question in the forest of similar elements. In this case, the fact that it is about a restaurant is not the important part, but that it is a shop.

Reply:
Anyway, I acknowledge the argument about the picnic table. Let's please discuss this in another ticket or discussion and collect there, which more elements should be shown if we were to soften the "show only same elements" guideline.

Reply:
> Let's please discuss this in another ticket or discussion and collect there, which more elements should be shown if we were to soften the "show only same elements" guideline.

See #4020!